internal/rbd/rbd_util.go

```diff
@@ -825,47 +811,30 @@ func getSnapshotMetadata(ctx context.Context, pSnapOpts *rbdSnapshot, cr *util.C
 	return nil
 }
 
-// imageInfo strongly typed JSON spec for image info
-type imageInfo struct {
-	ObjectUUID string   `json:"name"`
-	Size       int64    `json:"size"`
-	Features   []string `json:"features"`
-	CreatedAt  string   `json:"create_timestamp"`
-}
-
 // getImageInfo queries rbd about the given image and returns its metadata, and returns
 // ErrImageNotFound if provided image is not found
-func getImageInfo(ctx context.Context, monitors string, cr *util.Credentials, poolName, imageName string) (imageInfo, error) {
-	// rbd --format=json info [image-spec | snap-spec]
-
-	var imgInfo imageInfo
+func (rv *rbdVolume) getImageInfo() error {
+	image, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer image.Close()
 
-	stdout, stderr, err := util.ExecCommand(
-		"rbd",
-		"-m", monitors,
-		"--id", cr.ID,
-		"--keyfile="+cr.KeyFile,
-		"-c", util.CephConfigPath,
-		"--format="+"json",
-		"info", poolName+"/"+imageName)
+	imageInfo, err := image.Stat()
 	if err != nil {
-		klog.Errorf(util.Log(ctx, "failed getting information for image (%s): (%s)"), poolName+"/"+imageName, err)
-		if strings.Contains(string(stderr), "rbd: error opening image "+imageName+
-			": (2) No such file or directory") {
-			return imgInfo, ErrImageNotFound{imageName, err}
-		}
-		return imgInfo, err
+		return ErrImageNotFound{err: errors.Wrapf(err, "failed to get stat of image")}
```

Comment:
same here as well.

Reply:
it is most likely some other error, as opening the image already worked. But the comment for this function mentions it should return ErrImageNotFound (although if rv.open() fails, it could be util.ErrPoolNotFound)

Reply:
my question was what if image.Stat returns some other error may be internal server and got able to get stat of the image currently or something. because once we return ErrImageNotFound will delete data in omap and can leave stale image in backend

Reply:
@nixpanic i was waiting for the response of this comment. if we return any error from stats as ImageNotFound can cause serious issues that may delete omap entry and leave stale backend images, based on some condition if we can treat some error as not found and return it as not found makes sense.

Reply:
Ah, ok, I see! Will adjust the returning of the errors. `open()` already translates the error to `ErrImageNotFound`, and that is the only place where this would get returned.

Reply:
This should be addressed now. Please check again, @Madhu-1